core/src/unprocessed_packet_batches.rs

```diff
@@ -368,9 +374,40 @@ pub fn packet_message(packet: &Packet) -> Result<&[u8], DeserializedPacketError>
         .ok_or(DeserializedPacketError::SignatureOverflowed(sig_size))
 }
 
-/// Computes `(addition_fee + base_fee / requested_cu)` for `deserialized_packet`
-fn compute_fee_per_cu(_message: &VersionedMessage, _bank: &Bank) -> u64 {
-    1
+/// Computes `(addition_fee / requested_cu)` if feature tx_wide_compute_cap is enabled,
+/// otherwise return default 0u64
+fn compute_fee_per_cu(message: &VersionedMessage, bank: &Bank) -> u64 {
+    if bank.feature_set.is_active(&tx_wide_compute_cap::id()) {
+        if let Some(sanitized_message) = sanitize_message(message, bank) {
```

Comment:
And, still needs a `bank` as `address_loader` to sanitize message. Is there a way to do the same without needing a bank?

Reply:
From my investigation into the address loader here: https://github.com/solana-labs/solana/pull/25007/files#r866276411, it doesn't seem like it makes sense to sanitize the message until we get to actually processing the transaction within the leader bank, because sanitize could return different results based on which bank is being used to lookup the account. 

Double checking this is true with @jstarry

Reply:
I think we should start charging for all of these errors, AccountNotFound, InsufficientFundsForFee, InvalidAccountForFee instead of ignoring them

Then we don't need to sanitize.

Reply:
on the same page. For the context, the reason to sanitize is so compute_budget can process the sanitized message.

Reply:
sanitize for `v0::Message`s can incur multiple account lookups to resolve address maps. putting this before tx scheduling may introduce a DOS vector.

further, as Carl mentioned, `v0::Message`s expect address maps to be resolved and the transaction executed on the the same `Bank`, we no longer have that guarantee with this change.

Justin is going to work on requiring that top-level instruction program ids be static addresses, declared directly in the transaction rather than via address maps.  This will allow us to defer address map resolution to transaction accounts load and resolve ComputeBudget instructions via static analysis.

Reply:
My PRs are here:
https://github.com/solana-labs/solana/pull/25035
https://github.com/solana-labs/solana/pull/25041

Reply:
@taozhu-chicago you can rebase off of #25041 to use `SanitizedVersionedMessage` here

Reply:
awesome, cherry-picked

Reply:
commit 5e43496762a9421799626318aaf805a4fb888bf2 uses `SanitizedVersionedMessage` to process compute_budget, without needing a `bank`! Awesome ... thanks @jstarry